### PR TITLE
chore(tests): Add basic test for k-random nodes

### PR DESCRIPTION
**Pull request by ahsanbarkati**
Add some sanity tests for k-random nodes argument.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7758)

### CODE REVIEW

File: query/query0_test.go [hunk 1]
```diff
@@ -3484,6 +3484,38 @@ func TestMatchingWithPagination(t *testing.T) {
 	}
 }
 
+func TestKRandomNodes(t *testing.T) {
+	q := `{
+		data(func: uid(61, 62, 63, 64, 65, 66, 67, 68, 69, 70), random: 2) @filter(has(connects)) {
```
Comment (NamanJain8) [inline query/query0_test.go:3489]: Node 69,70 exist?

File: query/query0_test.go [hunk 2]
```diff
@@ -3484,6 +3484,38 @@ func TestMatchingWithPagination(t *testing.T) {
...
+			]},
+		{"kname":"can_be_picked",
+		"connects":[
+			{"kname":"yes"},
+			{"kname":"yes"}
+		]}]}}`
+	require.JSONEq(t, expected, result)
+
+	q = `{
+		data(func: uid(61, 62, 63, 64, 65, 66, 67, 68, 69, 70), random: 10) @filter(has(connects)) {
```
Comment (NamanJain8) [inline query/query0_test.go:3511]: What if number of input uids is less than 10?

**Review by NamanJain8 — APPROVED** [review]
Looks good to me. Just a couple of questions.

File: query/query0_test.go (hunk 1 above)
Comment (ahsanbarkati) [reply query/query0_test.go]: Yes, they exist but It doesn't matter, but for clarity, I have removed them from the query.

File: query/query0_test.go (hunk 2 above)
Comment (ahsanbarkati) [reply query/query0_test.go]: It was just by chance that, number of nodes in the query and the value of random was chosen to be the same. No specific reason.

**Pull request merged**